cloud.foundry.cli/src/main/java/cloud/foundry/cli/logic/ApplyLogic.java

```diff
@@ -51,31 +58,68 @@ public void applyApplications(@Nonnull Map<String, ApplicationBean> desiredAppli
         ConfigBean desiredApplicationsConfig = createConfigFromApplications(desiredApplications);
         ConfigBean liveApplicationsConfig = createConfigFromApplications(liveApplications);
 
         // compare entire configs as the diff wrapper is only suited for diff trees of these
-        DiffLogic diffLogic = new DiffLogic();
         log.info("Comparing the applications...");
-        DiffResult wrappedDiff = diffLogic.createDiffResult(liveApplicationsConfig, desiredApplicationsConfig);
+        DiffResult diffResult = this.diffLogic.createDiffResult(liveApplicationsConfig, desiredApplicationsConfig);
         log.info("Applications compared.");
 
-        Map<String, List<CfChange>> allApplicationChanges = wrappedDiff.getApplicationChanges();
-        ApplicationsOperations appOperations = new ApplicationsOperations(cfOperations);
+        Map<String, List<CfChange>> allApplicationChanges = diffResult.getApplicationChanges();
 
         Flux<Void> applicationRequests = Flux.fromIterable(allApplicationChanges.entrySet())
-                .flatMap( appChangeEntry -> ApplicationRequestsPlaner.create(appOperations, appChangeEntry.getKey(),
+                .flatMap( appChangeEntry -> ApplicationRequestsPlaner.createApplyRequests(applicationsOperations,
+                        appChangeEntry.getKey(),
                         appChangeEntry.getValue()))
-                .onErrorContinue(log::warning);
+                .onErrorContinue(log::error);
         applicationRequests.blockLast();
 
         log.info("Applying changes to applications...");
     }
 
+    //TODO update the documentation as soon as the method does more than just creating/removing services
+    /**
+     * Creates/removes all provided services.
+     * @param desiredServices the applications that should all be present in the live system after the procedure
+     * @throws ApplyException if an error occurs during the procedure
+     * @throws NullPointerException if the argument is null
+     */
+    public void applyServices(@Nonnull Map<String, ServiceBean> desiredServices) {
+        checkNotNull(desiredServices);
+
+        ServicesOperations servicesOperations = new ServicesOperations(cfOperations);
+        log.info("Fetching information about services...");
+        Map<String, ServiceBean> liveServices = this.getLogic.getServices(servicesOperations);
+        log.info("Information fetched.");
+
+        // that way only the applications of the live system are compared in the diff
+        ConfigBean desiredServicesConfig = createConfigFromServices(desiredServices);
+        ConfigBean liveServicesConfig = createConfigFromServices(liveServices);
+
+        // compare entire configs as the diff wrapper is only suited for diff trees of these
+        log.info("Comparing the services...");
+        DiffResult diffResult = this.diffLogic.createDiffResult(liveServicesConfig, desiredServicesConfig);
+        log.info("Services compared.");
```

Comment:
This line qualifies for "verbose" more than for "info".

Reply:
i aggree